### ISSUE TITLE
Review `RariFuseAllocator`

### ISSUE THREAD

**Issue by PARABRAHMAN0** [OlympusDAO/olympus-contracts#268]
Below I am giving you a link to the contract which should be reviewed including tests.

https://github.com/OlympusDAO/olympus-contracts/blob/570185d9e820d6bb7cb2a252a9a3049cba7b49a5/contracts/allocators/RariFuseAllocator.sol#L28

https://github.com/OlympusDAO/olympus-contracts/blob/570185d9e820d6bb7cb2a252a9a3049cba7b49a5/test/allocators/RariFuseAllocator.test.ts#L124

https://github.com/OlympusDAO/olympus-contracts/blob/570185d9e820d6bb7cb2a252a9a3049cba7b49a5/scripts/typescript/deployRariFuseAllocator.ts#L17

**Comment by PARABRAHMAN0** [comment]
Yes, in the future I will work on a local branch and issue this as a PR. Up until now I have been doing this on main. I will keep pushing reorganization though directly to main.

**Comment by 0xLienid** [comment]
> IERC20Metadata b = IERC20Metadata(address(_tokens[index]));
any reason for using IERC20Metadata vs IERC20 given you just use balanceOf and approve?

also, is there a better variable name option than "b"

**Comment by 0xLienid** [comment]
`
 if (balance > 0) {
       if (amounts[i] == type(uint256).max) f.redeem(f.balanceOf(address(this)));
`

looks like you already cache f.balanceOf(address(this)) to the balance variable, can just use here.

**Comment by ind-igo** [comment]
> any reason for using IERC20Metadata vs IERC20 given you just use balanceOf and approve? also, is there a better variable name option than "b"

I think this is because he needs the decimals in the `_worth` function

**Comment by ind-igo** [comment]
logic looks good to me
